rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -191,6 +191,12 @@ void Player::play()
       "Invalid delay value: " << play_options_.delay.nanoseconds() << ". Delay is disabled.");
   }
 
+  // If a non-default (positive) starting time is provided in PlayOptions,
+  // override the starting time obtained from reader metadata
+  if (play_options_.starting_time > INT64_C(0)) {
+    starting_time_ = play_options_.starting_time;
```

Comment:
> how - in general - PlayOptions::starting_time and BagMetadata::starting_time reside together - do they represent the same information?

I think that this option makes most sense as an "offset from the starting_time of the bag". `BagMetadata::starting_time` means "the timestamp of the first message in the bag" (though there is some discussion of changing it to mean "the timestamp at the time recording was started" - that is not the meaning presently). This value can be anything, as it depends on the recording system's steady clock.

However, the user likely does not care what the "absolute time" for starting is. E.g. if a bag has `starting_time=100`, and the user wants to start 20 seconds into the bag, they probably want to say `--start-time 20`, not `--start-time 120`, since the latter requires extra knowledge about the specific bag.

Given that, I would be more in favor this line looking like
```suggestion
    starting_time_ = metadata.starting_time + play_options_.start_offset;
```

Reply:
> I think that this option makes most sense as an "offset from the starting_time of the bag". 

Yes, that's exactly how I expect this to be used. I concur that the user would mean play from 20 seconds into the bag when they will use `--start-time 20`, so I will incorporate this suggested change.

On a related note, where should error checking for the `play_options_.starting_time` go? Does it make sense to throw a runtime error if `play_options_.starting_time < 0` here? Or should an error/warning be raised at construction/assignment time? Check https://github.com/ros2/rosbag2/pull/931#discussion_r763526466 for more context.

Reply:
I added the `start_offset` to `starting_time` in the `Player` constructor rather than the `play` function - primarily because that means we only add to the `starting_time` only once. From that point onwards, it doesn't matter if an offset was specified in `play_options_`, I think.

Check https://github.com/ros2/rosbag2/pull/931/commits/c79d65cfacb565d6c8b7829ba2bdefce5d61cac0 for reference.